src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -566,12 +566,15 @@ private static string NormalizeSearchPattern(string searchPattern)
                 return searchPattern;
             }
 
-            private static Microsoft.Win32.SafeHandles.SafeDirectoryHandle OpenDirectory(string fullPath)
+            private static Microsoft.Win32.SafeHandles.SafeDirectoryHandle OpenDirectory(string fullPath, bool throwOnError = true)
             {
                 Microsoft.Win32.SafeHandles.SafeDirectoryHandle handle = Interop.Sys.OpenDir(fullPath);
                 if (handle.IsInvalid)
                 {
-                    throw Interop.GetExceptionForIoErrno(Interop.Sys.GetLastErrorInfo(), fullPath, isDirectory: true);
+                    if (throwOnError)
+                        throw Interop.GetExceptionForIoErrno(Interop.Sys.GetLastErrorInfo(), fullPath, isDirectory: true);
+                    else
+                        return null;
```

Comment:
Doesn't this change the behavior from the Windows implementation?  For example, what if the directory exists but we don't have access to it?